src/Illuminate/Database/DatabaseManager.php

```diff
@@ -151,7 +151,7 @@ protected function configuration($name)
             throw new InvalidArgumentException("Database [{$name}] not configured.");
         }
 
-        return $config;
+        return $this->app->make(UrlParser::class)->parseDatabaseConfigWithUrl($config);
```

Comment:
You don't need to resolve the UrlParser here since it doesn't have any dependencies.

```suggestion
        $urlParser = new UrlParser;

        return $urlParser->parseDatabaseConfigWithUrl($config);
```